wp-content/themes/core/components/blocks/gallery_grid/Gallery_Grid_Controller.php

```diff
@@ -0,0 +1,421 @@
+		return defer_template_part( 'components/image/image', null,
+			[
+				Image_Controller::IMG_ID       => $img_id,
+				Image_Controller::AS_BG        => false,
+				Image_Controller::USE_LAZYLOAD => false,
+				Image_Controller::SRC_SIZE     => Image_Sizes::CORE_FULL,
+				Image_Controller::SRCSET_SIZES => [
+					'medium',
+					'medium_large',
+					'large',
+					Image_Sizes::CORE_FULL,
+				],
+			],
+		);
+	}
+
+	/**
+	 * @param int $index
+	 *
+	 * @return Deferred_Component
+	 */
+	protected function gallery_count( int $index ): Deferred_Component {
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::CLASSES => [
+				'b-gallery-grid__meta-count',
+			],
+			Text_Controller::CONTENT => sprintf( __( '%d of %d', 'tribe' ), $index + 1, count( $this->get_gallery_img_ids() ) ),
+		] );
+	}
+
+	/**
+	 * @param int $slide_id
+	 *
+	 * @return string|Deferred_Component
+	 */
+	protected function get_image_caption( int $slide_id ) {
+		$thumbnail_image = get_posts( [ 'p' => $slide_id, 'post_type' => 'attachment' ] );
+
+		if ( empty( $thumbnail_image ) && empty( $thumbnail_image[0] ) ) {
+			return '';
+		}
+
+		return defer_template_part( 'components/text/text', null, [
+			Text_Controller::CLASSES => [
+				'b-gallery-grid__meta-caption',
+			],
+			Text_Controller::CONTENT => esc_html( $thumbnail_image[0]->post_excerpt ) ?? '',
+		] );
+	}
+
+	/**
+	 * @param $img_id
+	 *
+	 * @return string
+	 */
+	protected function get_image_template( $index, $img_id ): string {
+		$img          = $this->get_slide_img( $img_id );
+		$slide_markup = $img;
+		$slide_markup .= defer_template_part( 'components/container/container', null, [
+			Container_Controller::CLASSES => [
+				'b-gallery-grid__meta-wrap',
+			],
+			Container_Controller::CONTENT => $this->gallery_count( $index ) . $this->get_image_caption( $img_id ),
+		] );
+
+		return $slide_markup;
+	}
+
+	/**
+	 * Get Slides
+	 *
+	 * @return array
+	 */
+	protected function get_slides(): array {
+		$slide_ids = $this->get_gallery_img_ids();
+		$slides    = [];
+
+		if ( empty( $slide_ids ) ) {
+			return $slides;
+		}
+
+		return array_map( function ( $index, $slide_id ) {
+			return $this->get_image_template( $index, $slide_id );
+		}, array_keys($slide_ids), $slide_ids );
+	}
+
+	/**
+	 * @return array
+	 */
+	public function get_gallery_img_thumbs(): array {
+		$ids          = $this->get_gallery_img_ids();
+		$gallery_imgs = [];
+		$i            = 1;
+		$img_size     = Image_Sizes::SQUARE_MEDIUM;
+		$img_bg       = true;
+		$img_bg_class = 'c-image--bg';
+		$img_aspect   = 's-aspect-ratio-1-1';
+		$img_srcset   = [
+			Image_Sizes::SQUARE_MEDIUM,
+			Image_Sizes::SQUARE_XSMALL
+		];
+
+		if ( empty( $ids ) ) {
+			return $gallery_imgs;
+		}
+
+		if ( $this->grid_layout === self::ONE ) {
+			$img_size     = Image_Sizes::CORE_FULL;
+			$img_bg       = false;
+			$img_bg_class = '';
+			$img_aspect   = '';
+			$img_srcset   = [ 
+				'medium',
+				'medium_large',
```

Comment:
Are `medium` and `medium_large` actual WP Core sizes? I wasn't aware that they existed.